includes/class-activity-dispatcher.php

```diff
@@ -30,48 +33,48 @@ public static function init() {
 	/**
 	 * Send Activities to followers and mentioned users or `Announce` (boost) a blog post.
 	 *
-	 * @param WP_Post $wp_post The ActivityPub Post.
-	 * @param string  $type    The Activity-Type.
+	 * @param mixed  $wp_object The ActivityPub Post.
+	 * @param string $type      The Activity-Type.
 	 *
 	 * @return void
 	 */
-	public static function send_activity_or_announce( WP_Post $wp_post, $type ) {
+	public static function send_activity_or_announce( $wp_object, $type ) {
 		// check if a migration is needed before sending new posts
 		Migration::maybe_migrate();
 
 		if ( is_user_type_disabled( 'blog' ) ) {
 			return;
 		}
 
-		$wp_post->post_author = Users::BLOG_USER_ID;
+		$wp_object->post_author = Users::BLOG_USER_ID;
 
 		if ( is_single_user() ) {
-			self::send_activity( $wp_post, $type );
+			self::send_activity( $wp_object, $type );
 		} else {
-			self::send_announce( $wp_post, $type );
+			self::send_announce( $wp_object, $type );
 		}
 	}
 
 	/**
 	 * Send Activities to followers and mentioned users.
 	 *
-	 * @param WP_Post $wp_post The ActivityPub Post.
-	 * @param string  $type    The Activity-Type.
+	 * @param mixed  $wp_object The ActivityPub Post.
+	 * @param string $type      The Activity-Type.
 	 *
 	 * @return void
 	 */
-	public static function send_activity( WP_Post $wp_post, $type ) {
-		if ( is_user_disabled( $wp_post->post_author ) ) {
+	public static function send_activity( $wp_object, $type ) {
+		if ( is_user_disabled( $wp_object->post_author ) ) {
 			return;
 		}
 
-		$object = Post::transform( $wp_post )->to_object();
+		$object = Factory::get_transformer( $wp_object )->to_object();
 
 		$activity = new Activity();
 		$activity->set_type( $type );
 		$activity->set_object( $object );
 
-		$follower_inboxes  = Followers::get_inboxes( $wp_post->post_author );
+		$follower_inboxes  = Followers::get_inboxes( $wp_object->post_author );
```

Comment:
We will need a better check here since the comment->post_author isn't guaranteed to be the same user as the comment author

Reply:
good point! what do you think about transforming the object before scheduling it? so that the dispatcher already get the Activity Object?

Reply:
but this seems to be a bigger change... let's try to fix this first...